tests/expectations/metrics/test_metrics_util.py

```diff
@@ -272,3 +295,149 @@ def test_get_unexpected_indices_for_multiple_pandas_named_indices_named_unexpect
+    if confirm_not_equal_column_name:
+        assert metric_domain_kwargs["column"] != confirm_not_equal_column_name
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "input_column_name_a, input_column_name_b, output_column_name_a, output_column_name_b",
+    [
+        [
+            "ARTISTS",
+            sqlalchemy.quoted_name(value="travel_agents", quote=True),
+            "ARTISTS",
+            sqlalchemy.quoted_name(value="travel_agents", quote=True),
+        ],
+        [
+            "ARTISTS",
+            "travel_agents",
+            "ARTISTS",
+            sqlalchemy.quoted_name(value="travel_agents", quote=True),
+        ],
+    ],
+)
+def test_get_dbms_compatible_metric_domain_column_pair_kwargs(
+    sa,
+    column_names_all_uppercase: list[str],
+    input_column_name_a: str,
+    input_column_name_b: str,
+    output_column_name_a: Union[str, sqlalchemy.quoted_name],
+    output_column_name_b: Union[str, sqlalchemy.quoted_name],
+):
+    quoted_column_name: sqlalchemy.quoted_name = sqlalchemy.quoted_name(
+        value="travel_agents", quote=True
+    )
+    test_column_names: list[str] = column_names_all_uppercase + [quoted_column_name]
+
+    metric_domain_kwargs: dict
+
+    metric_domain_kwargs = get_dbms_compatible_metric_domain_kwargs(
+        metric_domain_kwargs={
+            "column_A": input_column_name_a,
+            "column_B": input_column_name_b,
+        },
+        batch_columns_list=test_column_names,
+    )
+    assert metric_domain_kwargs["column_A"] == output_column_name_a
+    assert metric_domain_kwargs["column_B"] == output_column_name_b
+
+
+@pytest.mark.unit
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "input_column_list, output_column_list",
+    [
+        [
+            ["ARTISTS", "travel_agents", "FarmAnimals", "Household_Pets"],
+            [
+                "ARTISTS",
+                sqlalchemy.quoted_name(value="travel_agents", quote=True),
+                sqlalchemy.quoted_name(value="FarmAnimals", quote=True),
+                sqlalchemy.quoted_name(value="Household_Pets", quote=True),
+            ],
+        ],
+    ],
+)
+def test_get_dbms_compatible_metric_domain_column_list_kwargs(
+    sa,
+    column_names_all_uppercase: list[str],
+    input_column_list: list[str],
+    output_column_list: list[Union[str, sqlalchemy.quoted_name]],
+):
+    quoted_column_name_0: sqlalchemy.quoted_name = sqlalchemy.quoted_name(
+        value="travel_agents", quote=True
+    )
+    quoted_column_name_1: sqlalchemy.quoted_name = sqlalchemy.quoted_name(
+        value="FarmAnimals", quote=True
+    )
+    quoted_column_name_2: sqlalchemy.quoted_name = sqlalchemy.quoted_name(
+        value="Household_Pets", quote=True
+    )
+    test_column_names: list[str] = column_names_all_uppercase + [
+        quoted_column_name_0,
+        quoted_column_name_1,
+        quoted_column_name_2,
+    ]
+    random.shuffle(test_column_names)
```

Comment:
Consider deleting this shuffle since we are sorting anyway.

Reply:
@anthonyburdi This shuffle is needed here in order to intersperse the input column names to insure that the algorithm does not depend on any position.  The sorting at the end is to ensure that the types are correct, regardless of the order.  Happy to discuss.  Thanks.

Reply:
@anthonyburdi -- absolutely -- in next commit.  Thanks!